File: gameserver/src/main/java/l2s/gameserver/skills/effects/p_block_chat.java
package l2s.gameserver.skills.effects;

import l2s.gameserver.model.Creature;
import l2s.gameserver.model.Skill;
import l2s.gameserver.model.actor.instances.creature.Abnormal;
import l2s.gameserver.templates.skill.EffectTemplate;

public class p_block_chat extends Abnormal
{
	public p_block_chat(Creature creature, Creature target, Skill skill, boolean reflected, EffectTemplate template)
	{
		super(creature, target, skill, reflected, template);
	}

	@Override
	public boolean checkCondition()
	{
		return getEffected().isPlayer() && getTemplate().checkCondition(this);
	}

	@Override
	public void onStart()
	{
		super.onStart();
		getEffected().getPlayer().startChatBlock();
	}

	@Override
	public void onExit()
	{
		super.onExit();
		getEffected().getPlayer().stopChatBlock();
	}
}

File: gameserver/src/main/java/l2s/gameserver/handler/effects/impl/pump/retail/p_block_chat.kt
package l2s.gameserver.handler.effects.impl.pump.retail

import l2s.gameserver.handler.effects.impl.AbstractBooleanStatEffect
import l2s.gameserver.model.Creature
import l2s.gameserver.model.actor.instances.creature.Abnormal
import l2s.gameserver.stats.BooleanStat
import l2s.gameserver.templates.skill.EffectTemplate

/**
 * Block Chat effect implementation.
 * @author BiggBoss
 * @author Java-man
 *
 * @since 19.10.2019
 */
class p_block_chat(template: EffectTemplate) :
        AbstractBooleanStatEffect(template, BooleanStat.BLOCK_CHAT) {

    override fun checkPumpCondition(abnormal: Abnormal?, caster: Creature, target: Creature): Boolean {
        return target.isPlayer
    }

}